nicegui/elements/leaflet.js

```diff
@@ -0,0 +1,44 @@
+export default {
+  template: "<div></div>",
+  props: {
+    map_options: Object,
+  },
+  async mounted() {
+    await this.load_dependencies();
+    this.map = L.map(this.$el, this.map_options);
+    this.map.on("moveend", (e) => this.$emit("moveend", e.target.getCenter()));
+    this.map.on("zoomend", (e) => this.$emit("zoomend", e.target.getZoom()));
+    const connectInterval = setInterval(async () => {
+      if (window.socket.id === undefined) return;
+      this.$emit("init", { socket_id: window.socket.id });
+      clearInterval(connectInterval);
+    }, 100);
+  },
+  updated() {
+    this.map.setView(L.latLng(this.map_options.center.lat, this.map_options.center.lng), this.map_options.zoom);
+  },
+  methods: {
+    load_dependencies() {
+      if (!document.querySelector(`style[data-leaflet-css]`)) {
+        const link = document.createElement("link");
+        link.setAttribute("href", "https://unpkg.com/leaflet@1.6.0/dist/leaflet.css");
+        link.setAttribute("rel", "stylesheet");
+        link.setAttribute("data-leaflet-css", "");
+        document.head.appendChild(link);
+      }
+      if (!document.querySelector(`script[data-leaflet-js]`)) {
+        const script = document.createElement("script");
+        script.setAttribute("src", "https://unpkg.com/leaflet@1.6.0/dist/leaflet.js");
```

Comment:
I think we should serve the files locally like we do for the other elements to allow offline-functionality. Or is that unarchivable due to online-loaded maps material?

Reply:
I tried registering leaflet.js as a dependency. But our current system doesn't support additional assets like leaflet.css and marker icons. After struggling a while, I decided to use online resources for now and think about offline-functionality later.

Is there a use case for an offline Leaflet map? Would we support caching tiles for certain regions? Or would someone like to use a map without any tiles, just with vector annotations and maybe individual geo-referenced images?

Reply:
Ok, offline-functionality can be added later. I think it would be great to have the ability to provide tiles for certain regions. But that's quite complicated I guess. Let's postpone it.